lib/Models/ApiTableCatalogItem.ts

```diff
@@ -0,0 +1,197 @@
+            .reduce((curr, prev) => curr.concat(prev), []) // flatten
+            // make id/data pair for perIdData map
+            .map(data => [data[this.idKey!], data])
+        );
+
+        // Merge all responses with the same id (and timestamp, if applicable),
+        // Each of these merged responses will become a row in the table.
+        const perRowData = values
+          .filter(val => val.api.kind === "PER_ROW")
+          .map(val => val.data)
+          .reduce((curr, prev) => curr.concat(prev), []);
+
+        // Group all responses wth the same id
+        const groupedData = groupBy(perRowData, this.idKey);
+
+        this.apiResponses = Object.keys(groupedData).map(key => {
+          // Then, merge all responses with that key into a single object
+          const mergedResponse = groupedData[key].reduce(
+            (prev, curr) => Object.assign(prev, curr),
+            {}
+          );
+
+          // Add per id data for this response
+          Object.assign(
+            mergedResponse,
+            perIdData.get(mergedResponse[this.idKey!])
+          );
+          return mergedResponse;
+        });
+      });
+    });
+  }
+
+  protected makeTableColumns(addHeaders: boolean) {
+    return this.apis
+      .map(api =>
+        api.keyToColumnMapping.map(mapping =>
+          addHeaders ? [mapping.columnName!] : []
+        )
+      )
+      .reduce((prev, curr) => prev.concat(curr), []);
+  }
+
+  protected apiResponseToTable() {
+    const columnMajorTable = this.makeTableColumns(!this.hasData);
+
+    if (!this.apiDataIsLoaded) {
+      // No data yet, just return the headers
+      return columnMajorTable;
+    }
+    // Fill in column values from the API response
+    let startColumnIdx = 0; // Store where the columns for each API start
+    this.apis.forEach(api => {
+      this.apiResponses.forEach(response => {
+        api.keyToColumnMapping.forEach((mapping, mappingIdx) => {
+          // Get cell value
+          let cellValue = response[mapping.keyInApiResponse!];
+          if (cellValue === undefined) {
+            cellValue = "";
+          }
+          // Append the new value to the correct column
+          const columnIdx = startColumnIdx + mappingIdx;
+          columnMajorTable[columnIdx].push(`${cellValue}`);
+        });
+      });
+      startColumnIdx += api.keyToColumnMapping.length;
+    });
+    return columnMajorTable;
+  }
+
+  protected async forceLoadMetadata(): Promise<void> {
+    return Promise.resolve();
+  }
+
+  protected async forceLoadTableData(): Promise<string[][] | undefined> {
+    return this.loadDataFromApis()
+      .then(() => {
+        runInAction(() => {
+          this.append(this.apiDataColumnMajor);
+          this.hasData = true;
+        });
+      })
+      .then(() => undefined);
+  }
+
+  refreshData(): void {
+    this.loadDataFromApis().then(() => {
+      runInAction(() => {
+        this.append(this.apiDataColumnMajor);
+      });
+    });
+  }
```

Comment:
As we talked about, this isn't proper usage of `@computed` to set `dataColumnMajor`, but until `TableMixin` allows catalog items to override `@computed dataColumnMajor` there isn't a good way to do this.

So - until `TableMixin` has been sorted - I think this should just call `this.apiResponseToTable()` instead of using a `@computed` here as `apiDataColumnMajor` will is going to be re-computed after every time `loadDataFromApis` is called